addons/beehave/nodes/leaves/blackboard_set.gd

```diff
@@ -0,0 +1,18 @@
+@tool
+class_name BlackboardSetAction extends ActionLeaf
+
+
+@export var key: String = ""
+@export var value: String = ""
+
+
+@onready var _expression: Expression = parse_expression(value)
+
+
+func tick(actor: Node, blackboard: Blackboard) -> int:
+	var result: Variant = _expression.execute([], blackboard)
+	if _expression.has_execute_failed():
+		return FAILURE
```

Comment:
Returning FAILURE here can be really confusing. We should always expect that the expression executes successfully. I think I will add assert or push_error here instead, to clearly inform the user that something went wrong.

Reply:
@Wichamir from a logical perspective, returning `SUCCESS` here would be wrong. Keeping it as it is makes sense to me. However, rather than logging or push error we should ask ourselves what could make the execution fail. I'd assume a wrong expression or invalid syntax of some sort? (reading the documentation suggests that https://docs.godotengine.org/en/latest/classes/class_expression.html#class-expression-method-execute)

In that case, we could even set up a configuration warning on these nodes in case someone setup a wrong expression syntax? What other reasons could there be for execute to fail?

Reply:
A configuration warning is a great idea! I can introduce it. This would eliminate syntax errors, but the problem still remains though. For instance:
`get_value("destination").rotated(PI)`
This is a perfectly valid syntax, but what if somehow the destination is not a Vector2, but a Vector3. Vector3 requires two arguments for the rotated method, so we get a runtime error.

Reply:
@bitbrain Okay so in the end I left only `return FAILURE` here, because it turns out `Expression` class itself pushes an error message anyway if something goes wrong.